src/schnetpack/representation/painn.py

```diff
@@ -161,8 +167,28 @@ def __init__(
         self.cutoff_fn = cutoff_fn
         self.cutoff = cutoff_fn.cutoff
         self.radial_basis = radial_basis
+        self.activate_charge_spin_embedding = activate_charge_spin_embedding
 
-        self.embedding = nn.Embedding(max_z, n_atom_basis, padding_idx=0)
+        # nuclear embedding layer (often complex nuclear embedding has negative impact on the performance of the model, so we can use simple nuclear embedding to avoid this issue)
+        if nuclear_embedding != "simple":
+            self.nuclear_embedding = NuclearEmbedding(self.n_atom_basis,max_z, zero_init=True)
+        else:
+            self.nuclear_embedding = nn.Embedding(max_z, self.n_atom_basis, padding_idx=0)
+
+        if self.activate_charge_spin_embedding:
```

Comment:
we could also pass directly pass the instances here and make them optional. The default could just be `None` here @jnsLs ?
also: does it make sense to only use charge or only use spin embedding? in this case it would also help to pass instantiated objects instead of `str`

Reply:
done in c86b404 and f99a432
I don't know how much sense it makes to call charge and spin embedding separately from each other or if they influence each other. As I have do not have the time right now to test this extensively, I would prefer to leave it as it is if this is ok for you.